src/support/metadata/TypeMetadata.ts

```diff
@@ -2,6 +2,37 @@ export class TypeMetadata {
     constructor(public target: Function,
                 public propertyName: string,
                 public sourcePropertyName: string,
-                public type: Function) {
+                private _type: Function) {
+    }
+
+    public get type(): Function
+    {
+        // If type name is empty, we will assume it returns a 'require'.
```

Comment:
just a small detail, but should this say `returns a resolver` – whether it's a `require` function or something else isn't really relevant from the library's perspective

Reply:
Could you also clarify _why_ type name would be empty? e.g
```
// Arrow functions do not have a name, while entity classes do, so when the
// name is empty, we will assume the user is passing a resolver function.
```

Reply:
Entity definitions are in `class Comment extends Entity` format and Typescript converts this into `function Comment() {}`. At this point `Comment.name` is `Comment` and the property is read-only.

_But_ since anonymous functions are well, anonymous, their `.name` prop is an empty string, but only if it's not in a _direct_ assignment expression. ([Further reading on function names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name))

```javascript
let foo = function () {}; // Direct assignment, foo.name is foo
let foo = (() => function () {})(); // Indirect assignment, function returns anonymous function, foo.name is empty string.
```

I'm emphasizing on this, because deferred types must only be passed the resolver function itself, and *must not* assign the resolver to a variable first and then pass it to type decorator. We should document that.

Also, assuming that this is a sufficient explanation, would you like me to add this to that comment block?

Reply:
sure, yeah, would be good to add – maybe have it in a test to demonstrate it?